mavros_extras/src/plugins/rpm.cpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * @brief RPM plugin
+ * @file rpm.cpp
+ * @author Pavlo Kolomiiets <pkolomiets@gmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2017 Pavlo Kolomiiets.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <mavros_msgs/RPM.h>
+
+namespace mavros {
+namespace extra_plugins {
+/**
+ * @brief Ardupilot rpm plugin.
+ *
+ * This plugin allows publishing RPM sensor data from Ardupilot FCU to ROS.
+ *
+ */
+class RPMPlugin : public plugin::PluginBase {
+public:
+	RPMPlugin() : PluginBase(),
+		nh("~")
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+
+		rpm_pub = nh.advertise<mavros_msgs::RPM>("rpm", 10);
+	}
+
+	Subscriptions get_subscriptions()
+	{
+		return {
+			make_handler(&RPMPlugin::handle_rpm)
+		};
+	}
+
+private:
+	ros::NodeHandle nh;
+
+	std::vector<float> rpms;
+
+	ros::Publisher rpm_pub;
+
+	/**
+	 * @brief Handle RPM MAVlink (Ardupilot) message.
+	 * Message specification: http://mavlink.org/messages/ardupilotmega#RPM
+	 * @param msg	Received Mavlink msg
+	 * @param rpm	RPM msg
+	 */
+	void handle_rpm(const mavlink::mavlink_message_t *msg, mavlink::ardupilotmega::msg::RPM &rpm) {
+
+		rpms.resize(2);
```

Comment:
If `rpm_msg->rpm` is a `float32[2]`, you don't need to create a vector and do this operation.

Reply:
Maybe in the future Ardupilot will report more rpm-s (4 wheel encoders, for example), so vector was chosen for extensibility.

Reply:
I think variable length is ok, but you do not need `rpms` member. Just use local variable, or do:
```cpp
rpm_msg->rpm.resize(2);
rpm_msg->rpm[0] = rpm.rpm1;
rpm_msg->rpm[1] = rpm.rpm2;
```

Reply:
@vooon Will update.